### PR TITLE
Add test for Halborn audit issue HAL-03 

### PR DESCRIPTION
**Pull request by vivekvpandya**
IMPROPER LENDING MARKET CONFIGURATION LEADS TO MULTIPLE VULNERABILITIES

## Issue
https://app.clickup.com/t/2953czv

## Description
Just adding a simple test showing the issue. 

## Checklist

~- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_~
~- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_~

### CODE REVIEW

File: frame/lending/src/tests.rs [hunk 1]
```diff
@@ -1181,6 +1181,62 @@ fn zero_amount_collateral_deposit() {
 	})
 }
 
+#[test]
+fn halborn_issue_market_owner_profits_by_lquidate() {
```
Comment (dzmitry-lahoda) [inline frame/lending/src/tests.rs:1185]: `lquidate` typo

Comment (dzmitry-lahoda) [inline frame/lending/src/tests.rs:1185]: may you add harbor_0003_... id here?

File: frame/lending/src/tests.rs [hunk 2]
```diff
@@ -1181,6 +1181,62 @@ fn zero_amount_collateral_deposit() {
...
+		let collateral_factor = MoreThanOneFixedU128::saturating_from_rational(2000_u128, 99_u128);
+		let updatable = UpdateInput {
+			collateral_factor,
+			under_collateralized_warn_percent: Percent::from_float(1.1),
+			liquidators: vec![],
+			interest_rate_model: InterestRateModel::default(),
+		};
+		let updatable = updatable.try_into_validated::<UpdateInputVaild>().unwrap();
+		assert_ok!(Lending::update_market(Origin::signed(*ALICE), market_id, updatable));
+		// ALICE loan must be liquidated now.
```
Comment (vimukthi-git) [inline frame/lending/src/tests.rs:1235]: Just to understand, Why would ALICE `update_market`/collateral factor to get their own loan liquidated?

Comment (vivekvpandya) [reply frame/lending/src/tests.rs:1235]: Sorry for confusion, ALICE is creator of the market so she can update the configs. This is just to show problematic case.

Comment (vivekvpandya) [reply frame/lending/src/tests.rs]: Updated test to avoid confusion.

**Review by vimukthi-git — APPROVED** [review]
Seems like a critical issue to fix no?

**Comment by vivekvpandya** [comment]
> Seems like a critical issue to fix no?

So its bit complicated issue and most likely we need some sort of governance mechanism , so perhaps will not be done in current sprint.

**Comment by vimukthi-git** [comment]
@vivekvpandya Then may be we should create a task already and prioritize no? Or is there any other process to handle this?

**Comment by hussein-aitlahcen** [comment]
Perhaps worth thinking about removing the ability to change the collateral factor for open markets and only allow the users to do so via closing + creating a new one?

**Comment by vivekvpandya** [comment]
This is ticket for https://app.clickup.com/t/2953d0u . Can we close this PR as it just intends to add test?

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]

**Review by vimukthi-git — APPROVED** [review]

**Comment by vimukthi-git** [comment]
Approved on the understanding that another team/dev is handling the fix for this.

**Pull request merged**